### ISSUE TITLE
Bundle and Rails Command Destination not found

### ISSUE THREAD

**Issue by anirudhy** [railsinstaller/railsinstaller-windows#70]
Bundle and Rails command are not working from CMD in 64 bits v3.2.0. I checked the bat file in the ruby directory and found that you added the static file destination for the ruby.exe.
Please fix that.

**Comment by dacoinminster** [comment]
I think I have the same problem. When I try to install the latest version of rails on windows I get a rails.bat which looks thinks ruby.exe is at:.
"C:\Users\emachnic\GitRepos\railsinstaller-windows\stage\Ruby2.2.0\bin\ruby.exe" 

I'm guessing "emachnic" is one of the devs that worked on this? 

Installing the older version of RailsInstaller fixed this problem.

**Comment by emachnic** [comment]
I'm currently testing a new patch version.

**Comment by emachnic** [comment]
I've got no idea why this is not working properly. For the time being, go ahead and reinstall the gems you need.

**Comment by sibinx7** [comment]
Change path in  bundle.bat and rails.bat file

**Comment by velaseriat** [comment]
Just reinforcing what was already said.

running "bundle" or "rails" gives me the following:

C:\RailsInstaller\Ruby2.2.0\bin>rails
The system cannot find the path specified.

And as mentioned above, I also see "emachnic" in a filepath bundle.bat. Guessing this would only work on your machine.

Thank you,
Velaseriat

**Comment by nullnvoid** [comment]
Are there any plans to fix this? 

RailsInstaller current Windows version is non functional when installed using the official package.

To help clarify, there are actually over a dozen batch files in the ruby2.2.0 bin folder that have hard coded paths to the stage folder on emachnic's personal pc.

**Comment by emachnic** [comment]
@nullnvoid Please submit a patch. I have no idea how to fix this right now

**Comment by avdi** [comment]
@emachnic I'm building RailsInstaller locally, and I can't repro the user-specific paths embedded in .bet files. For instance, my locally built `stage/Ruby2.0.0/bin/bundle.bat` contains:

```
@ECHO OFF
IF NOT "%~f0" == "~f0" GOTO :WinNT
ECHO.This version of Ruby has not been built with support for Windows 95/98/Me.
GOTO :EOF
:WinNT
@"%~dp0ruby.exe" "%~dpn0" %*
```

Can you rebuild and see if you still have your home directory embedded in the `bundle.bat` and `rails.bat` files?

If you're still seeing the issue, then assuming I'm able to build the final .exe I'd be happy to submit it for testing and deployment as a replacement for the current broken build.

**Comment by avdi** [comment]
Actually scratch that, I was on master. I do see the issue on the ruby2.2 branch.

**Comment by avdi** [comment]
My guess now is that this has something to do with a differing behavior in RubyGems on windows between Ruby 2.0 and 2.2.

**Comment by jpmermoz** [comment]
Just search for every *.bat file inside the C:\RailsInstaller\Ruby2.2.0\bin folder which contains "emachnic" paths, and replace the whole content of them with this code:

`@ECHO OFF
IF NOT "%~f0" == "~f0" GOTO :WinNT
ECHO.This version of Ruby has not been built with support for Windows 95/98/Me.
GOTO :EOF
:WinNT
@"%~dp0ruby.exe" "%~dpn0" %*
`

**Comment by avdi** [comment]
@jpmermoz the problem is that the files are generated this way during the build process. So it's going to crop up again with every build, until someone can get to the root of why the .bat files are generated that way.

**Comment by jpmermoz** [comment]
@avdi you're right.. I was just proposing a temporal solution for anyone having this issue after installing everything on windows

**Comment by emachnic** [comment]
@jpmermoz If you reinstall the offending gems, it should fix the path automatically.

@avdi Do we think this could be an upstream issue, possibly with Rubygems itself? Trying to figure out where this behavior cropped up.

**Comment by avdi** [comment]
I think it's likely a change in how RubyGems makes .bat stub files. I'm not ready to say it's bug, though. In the common case, generating stubs with hard-coded paths makes sense. I'm not sure they were ever intended to be redistributed like this.

**Comment by nullnvoid** [comment]
It looks like there's some kind of stage/production setting that has been
missed in the build process.  Hard coding in the dev environment makes
sense...not for deployment though.

**Comment by avdi** [comment]
I think the part that's unanticipated here is that binstub batch files
which are intended to be generated as part of final gem installation on a
specific system, are instead being bundled up and redistributed onto other
systems. I suspect the (reasonable) assumption that RubyGems makes is that
while a gem is redistributable, the files generated by `gem install` may
well be system and user-specific.
On Apr 11, 2016 8:40 PM, "nullnvoid" <notifications@github.com> wrote:

It looks like there's some kind of stage/production setting that has been
missed in the build process. Hard coding in the dev environment makes
sense...not for deployment though.

On 12 April 2016 at 08:47, Avdi Grimm <notifications@github.com> wrote:

> I think it's likely a change in how RubyGems makes .bat stub files. I'm
> not ready to say it's bug, though. In the common case, generating stubs
> with hard-coded paths makes sense. I'm not sure they were ever intended to
> be redistributed like this.

>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <
https://github.com/railsinstaller/railsinstaller-windows/issues/70#issuecomment-208598054
>

>

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/railsinstaller/railsinstaller-windows/issues/70#issuecomment-208644133>

**Comment by Syntaf** [comment]
Any updates on this? Trying to get into ruby on rails programming, but kinda tough when the go-to installer doesn't even work.

**Comment by Velaseriat** [comment]
If youre looking to get started, just go to an earlier version. 

http://github.com/railsinstaller/railsintaller-windows/release

Just keep going backwards until you find a version that doesnt have this error.

And remember: https://gist.github.com/fnichol/867550

If youre just starting out, youre not gonna miss out on much by going to an earlier version.

**Comment by michael-wise** [comment]
This bug is often followed by the windows ssl cert problem, one or both of which prevent an alarming number of those wanting to learn rails from even starting. I'm sure there are many who turn away and never chime in anywhere. I hope these entry-barrier bugs soon receive the community attention they probably deserve. Unfortunately, it is out of mine and many others expertise. So please let's all pass this plea around to friends who know how to cook on Windows.

**Comment by avdi** [comment]
This probably deserves a thread of its own, but:

I know this project must be a lot of effort to maintain. These days, as far
as I can tell, Chocolatey seems to work great for installing all the
necessities for Ruby on Rails development. Does it make sense to keep
offering a separate RailsInstaller?

I am *not* trying to denigrate the great work here. I'm just guessing that
nobody really loves maintaining this kind of stuff. Now that there's a
half-decent open source package manager on Windows, maybe that time could
be spent elsewhere?

**Comment by emachnic** [comment]
The advantage that RailsInstaller would offer is that it automatically sets
up the DevKit but really, you are very correct.

**Comment by emachnic** [comment]
Hi all, please download the latest version of RI with Ruby 2.2 and let me know if that resolves your issue.

**Comment by topiaryx** [comment]
@emachnic Still experiencing the same issue, had to edit _rails.bat_ and _bundle.bat_ in order to make this work. (This is with Ruby 2.3 on Windows 10 Enterprise - attempted install with elevated permissions)

**Comment by prusswan** [comment]
I just saw this when trying to upgrade my existing installation on Windows from Ruby 2.0 to 2.2. If the problem only lies with RailsInstaller but not RubyInstaller, then can I just install Ruby 2.2 using RubyInstaller?

The inferior/lack of Windows support is not good news, but RailsInstaller is still a popular tool for Windows users trying to learn Rails. Although, the advanced users will eventually move on to other platforms for more serious work.

**Comment by hkhatod** [comment]
INstalled 2.2 version on windows 10 64bit. No issues!!!

**Comment by mstramb** [comment]
I just downloaded and tried to install on Win7 https://s3.amazonaws.com/railsinstaller/Windows/railsinstaller-3.3.0.exe

Same "polluted" paths in all the bat files in C:\RailsInstaller2\Ruby2.3.0\bin

**Comment by MikroAcse** [comment]
Still relevant.

**Comment by kuolai** [comment]
Just replaced D:\RailsInstaller\Ruby2.3.0\bin\*.bat with the patch. There are 17 files.
It took some time to cut and paste. The good news is it works and I am good to go.
